copy_test.go

```diff
@@ -1118,3 +1167,104 @@ func TestStructToMap_SliceOfStructs_NonEmptyDst(t *testing.T) {
 		},
 	}, dst)
 }
+
+func TestStructToMap_EntireSlicePrimitive_NonEmptyDst(t *testing.T) {
+	type A struct {
+		Field1 []int
+	}
+	src := &A{
+		Field1: []int{1, 2, 4, 8},
+	}
+
+	dst := map[string]interface{}{
+		"Field1": []int{16, 32, 64},
+	}
+	mask := fieldmask_utils.MaskFromString("Field1")
+	err := fieldmask_utils.StructToMap(mask, src, dst)
+	require.NoError(t, err)
+	assert.Equal(t, map[string]interface{}{
+		"Field1": src.Field1,
+	}, dst)
+}
+
+func TestStructToMap_EntireSlice_NonEmptyDst(t *testing.T) {
+	type A struct {
+		Field1 string
+		Field2 string
+	}
+	type B struct {
+		A []A
+	}
+	src := &B{
+		A: []A{
+			{
+				Field1: "src ele1 field1",
+				Field2: "src ele1 field2",
+			},
+			{
+				Field1: "src ele2 field1",
+				Field2: "src ele2 field2",
+			},
+		},
+	}
+	dst := map[string]interface{}{
+		"A": []map[string]interface{}{
+			{
+				"Field1": "dst ele1 field1",
+			},
+			{
+				"Field2": "dst ele2 field 2",
+			},
+			{
+				"Field1": "dst ele3 field 3",
+			},
+		},
+	}
+	mask := fieldmask_utils.MaskFromString("A")
+	err := fieldmask_utils.StructToMap(mask, src, dst)
+	require.NoError(t, err)
+	assert.Equal(t, map[string]interface{}{
+		"A": []map[string]interface{}{
+			{
+				"Field1": src.A[0].Field1,
+				"Field2": src.A[0].Field2,
+			},
+			{
+				"Field1": src.A[1].Field1,
+				"Field2": src.A[1].Field2,
+			},
+		},
+	}, dst)
+}
+
+func TestStructToMap_EntireSlice_FailsWhenDestSliceLenIsLessThanSource(t *testing.T) {
```

Comment:
Hm... i would expect an existing dst slice to grow as needed to accommodate all the items from src.

There is a bug in the code (it was there before your modifications): we should treat arrays and slices differently (as in `structToStruct`) and grow dst slices if needed.
I will address this issue later today.
For

Reply:
cool, I was wondering why is it done different there, I wanted to change it to act like `StructToStruct`, but I thought it might be intended to be different for whatever reason ^^.